gamla/higher_order.py

```diff
@@ -34,14 +37,53 @@ def prepare_and_apply_async(f):
     14
     """
 
-    async def prepare_and_apply(value):
+    async def prepare_and_apply(value: Any):
         return await async_functions.to_awaitable(
             (await async_functions.to_awaitable(f(value)))(value),
         )
 
     return prepare_and_apply
 
 
+def ignore_first_arg(f: Callable) -> Callable:
+    """Ignores the first argument."""
+    if asyncio.iscoroutinefunction(f):
+
+        async def ignore_first_arg_async(_, *args, **kwargs):
+            return await f(*args, **kwargs)
+
+        return ignore_first_arg_async
+
+    def ignore_first_arg(_, *args, **kwargs):
+        return f(*args, **kwargs)
+
+    return ignore_first_arg
+
+
+def persistent_cache(
+    get_item: Callable[[str], Any],
+    set_item: Callable[[str, Any], None],
+    make_key: Callable[[Any], str],
+) -> Callable:
+    """Wraps a function with persistent cache. Gets the item getter and item setter as parameters."""
```

Comment:
let's add that if `f` is a coroutine we expect `get_item` and set_item to be coroutines as well.